linked_plots/callback_test.ipynb

```diff
@@ -0,0 +1,232 @@
+    "    good_indexes = np.logical_and(good_indexes, good_model_indexes)\n",
+    "    psf_mag = calib.getMagnitude(fluxes[good_indexes])\n",
+    "    cm_mag = calib.getMagnitude(model_fluxes[good_indexes])\n",
+    "    delta_mag = psf_mag - cm_mag\n",
+    "    extendedness = src['base_ClassificationExtendedness_value'][good_indexes]\n",
+    "    df = pd.DataFrame({'delta_mag':delta_mag, 'psf_mag':psf_mag, 'class':extendedness, 'visit':data_id['visit'], 'filter':data_id['filter']})\n",
+    "    return df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Loop over all visits (slow)\n",
+    "df = get_pandas_dataid(butler, data_ids[0])\n",
+    "for  i in range(1,len(data_ids)):\n",
+    "    df = pd.concat([df,get_pandas_dataid(butler,data_ids[i])])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(' Total objects = {}'.format(len(df)))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "xyarray = np.array([df.psf_mag.values, df.delta_mag.values]).T"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "np.shape(xyarray)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%opts Layout [ vspace=0.1 hspace=0.1]\n",
+    "%%opts Points [ tools=['hover', 'box_select' ,'lasso_select']](color='k')\n",
+    "# decimate create sampled data \n",
+    "Points = decimate(hv.Points(xyarray))\n",
+    "# use datashader\n",
+    "Points_shader = dynspread(datashade(hv.Points(xyarray)))\n",
+    "#selection will have the bounding box from the selection tools\n",
+    "selection = hv.streams.BoundsXY(source=Points)\n",
+    "\n",
+    "\n",
+    "def selected_hist(bounds):\n",
+    "    \"\"\"\n",
+    "    Generates a Holoview histogram based on the selection box.\n",
+    "    *Note* Do not change the bounds variable, which is an attribute \n",
+    "    of the stream selection object.\n",
+    "    \"\"\"\n",
+    "    arr = xyarray[:,0]\n",
+    "    if not bounds:\n",
+    "        lower = min(arr)\n",
+    "        upper = max(arr)\n",
+    "    else:\n",
+    "        lower = list(bounds)[0]\n",
+    "        upper = list(bounds)[2]\n",
+    "        \n",
+    "    sel = np.where((arr >= lower) & (arr < upper))\n",
+    "    frequencies, edges = np.histogram(arr[sel], 20)\n",
+    "    return hv.Histogram((edges, frequencies))\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# plots need some clean up and labels and better scaling for histogram\n",
+    "hv.Layout(Points +  Points_shader + hv.DynamicMap(selected_hist, streams=[selection],)).cols(2)\n",
```

Comment:
The Holoviews [linked stream](http://holoviews.org/user_guide/Custom_Interactivity.html) example looks great! however the selection did not work for me with the datashader. It must be a detail I can give a try tomorrow!